src/Spectre.Cli/Internal/Modelling/CommandParameter.cs

```diff
@@ -61,6 +61,31 @@ public void Assign(CommandSettings settings, object? value)
                 newArray.SetValue(value, newArray.Length - 1);
                 value = newArray;
             }
+
+            if (ParameterType.IsFlagValue())
+            {
+                var flagValue = (IFlagValue?)Property.GetValue(settings);
+                if (flagValue == null)
+                {
+                    var flagElementType = ParameterType.GetFlagElementType();
+                    flagValue = (IFlagValue?)Activator.CreateInstance(typeof(FlagValue<>).MakeGenericType(flagElementType));
+                    if (flagValue == null)
+                    {
+                        throw new InvalidOperationException("Could not create flag value.");
+                    }
+                }
+
+                if (value != null)
+                {
+                    // Null means set, but not with a valid value.
+                    flagValue.Value = value;
+                }
+
+                // If the parameter was mapped, then it's set.
+                flagValue.IsSet = true;
+
+                value = flagValue;
+            }

```

Comment:
Would like to start breaking this logic apart, but might perhaps be like it is for a little while more...